djangowind/auth.py

```diff
@@ -415,14 +482,41 @@ def ldap_lookup(uni=""):
 
 
 class CDAPProfileHandler(object):
-    """ fills in email, last_name, first_name from CDAP """
+    def __init__(self):
+        self._set_ldap_lookup()
+
+    def _ldap3_lookup(self, uni):
+        return ldap3_lookup(uni)
+
+    def _python_ldap_lookup(self, uni):
+        return python_ldap_lookup(uni)
+
+    def _set_ldap_lookup(self):
+        """ set the ldap lookup method based on what library is available """
+        # prefer ldap3
+        if ldap3 is not None:
+            self.ldap_lookup = self._ldap3_lookup
+            return
+
+        # fallback to python-ldap
+        if ldap is not None:
+            self.ldap_lookup = self._python_ldap_lookup
+            return
+
+        # neither are available
+        statsd.incr('djangowind.ldap_lookup.import_failed')
+        warn("""this requires a python ldap library.
+        you probably need to install 'ldap3', 'python-ldap' or
+        an equivalent""")
+
     def process(self, user):
+        """ fills in email, last_name, first_name from LDAP """
         statsd.incr('djangowind.cdap.called')
         if not user.email:
             user.email = user.username + "@columbia.edu"
         if not user.last_name or not user.first_name:
             try:
-                r = ldap_lookup(user.username)
+                r = self.ldap_lookup(user.username)
```

Comment:
Will this still result in an ImportError exception? self.ldap_lookup will just be undefined here.